library/src/scala/tasty/reflect/TreeUtils.scala

```diff
@@ -97,13 +99,25 @@ trait TreeUtils
       case TypeTree.Applied(tpt, args) => foldTypeTrees(foldTypeTree(x, tpt), args)
       case TypeTree.ByName(result) => foldTypeTree(x, result)
       case TypeTree.Annotated(arg, annot) => foldTree(foldTypeTree(x, arg), annot)
+      case TypeTree.TypeLambdaTree(typedefs, arg) => foldTrees(foldTypeTree(x, arg), typedefs)
+      case TypeTree.Bind(_, tbt) => foldTypeTree(x, tbt)
+      case TypeTree.Block(typedefs, tpt) => foldTrees(foldTypeTree(x, tpt), typedefs)
+      case TypeTree.MatchType(boundopt, selector, cases) => {
+        val bound_fold_result = boundopt.map(foldTypeTree(x, _)).getOrElse(x)
```

Comment:
`boundopt.fold(x)(foldTypeTree(x, _))` does the same without creating the intermediate object.

Reply:
Fixed